mlir/include/mlir/Interfaces/TilingInterface.td

```diff
@@ -131,6 +150,42 @@ def TilingInterface : OpInterface<"TilingInterface"> {
           return failure();
         }]
       >,
+      InterfaceMethod<
+        /*desc=*/[{
+          Method to generate the tiled implementation of an operation from
+          operand tile position.
```

Comment:
> why do we need this method that, from its description, is a trivial composition of those two?

I tried following the other comment regarding the same which now makes this as a static method instead. Does that work? The idea is to have a unified API which [tensor::replaceInsertSliceWithTiledConsumer](https://github.com/llvm/llvm-project/blob/8bde51492235e1bfe81b527414c19743ec987a62/mlir/lib/Dialect/Tensor/Transforms/SwapExtractSliceWithProducerPatterns.cpp#L58) can use.

> what guarantees do we have that an op doesn't end up with diverging implementations of these methods?

I'm not sure if I understood the question (moreover the intent) of this question. Did you mean to say what happens if two given ops are implementing these methods in an orthogonal way? In that case I'm failing to see how that's an issue if the implementation achieves what the method is aimed for.

Pinging @cxy-1993 in case he has pointers to field/add on to this as its based on the reverted commit.

Reply:
I think if we are doing this this should just be an `InterfaceMethod` with a `defaultImplementation`. I was seeing a similar thing for tile consumer + fuse producer. Most of the operations seem to do similar thing. I wasnt fully sure that this always works, so I hadnt implemented it, but it would make sense to have a default implementation that can be overwritten.
I dont think this should be a `StaticInterfaceMethod`.

Reply:
> I tried following the other comment regarding the same which now makes this as a static method instead. Does that work? The idea is to have a unified API which tensor::replaceInsertSliceWithTiledConsumer can use.

A "non-overridable" method would work. It's not be a static method (we have "static interfaces" that have "static methods" overridable by a specific subclass but not requiring an _instance_ of that subclass to execute). We have `extraClassDeclarations` and `extraSharedClassDeclarations` on interfaces that can be used for this purpose. (Don't take inspiration from the Linalg interface, it overuses overridable methods for things that shouldn't be overridable.)

> I'm not sure if I understood the question (moreover the intent) of this question. Did you mean to say what happens if two given ops are implementing these methods in an orthogonal way? In that case I'm failing to see how that's an issue if the implementation achieves what the method is aimed for.

The current design has overridable methods to do A->B, B->C and A->C. (where A is the operand tile, B is the iteration domain tile, and C is the tiled implementation). It is a dangerous design because the implementation that overrides them must guarantee that A->B->C and A->C produce the same result. And given the complexity of each of these functions this is far from being obvious to verify, let alone implement. So I am suggesting that A->C (this method) should not be overridable at all. 

There may be a case where there exists a more efficient implementation of A->C than doing A->B->C, which would justify making it overridable with a default implementation. I would like to see such a case before doing that.

Reply:
Sorry for the delay. 

My initial intention of implementing this interface was to provide a more concise interface for pass calls through the tiling interface. This situation often occurs in the linalg op interface and BuiltinTypeInterfaces. These interfaces have default implementations and usually do not require overloading when defining new ops. ftynse's opinion is also make sense, I am now more inclined to use StaticInterfaceMethod or extraClassDeclarations.

Reply:
Linalg op interface is clearly not the example to follow. Most of its methods should not be overridable.